File: src/main/java/com/github/zxh/classpy/classfile/bytecode/Multianewarray.java
package com.github.zxh.classpy.classfile.bytecode;

import com.github.zxh.classpy.classfile.ClassReader;
import com.github.zxh.classpy.classfile.jvm.Opcode;

public class Multianewarray extends Instruction {

    {
        u1  ("opcode");
        u2cp("index");
        u1  ("dimensions");
    }

    public Multianewarray(Opcode opcode, int pc) {
        super(opcode, pc);
    }
    
    @Override
    protected void readOperands(ClassReader reader) {
        setDesc(getDesc() + " "
                + super.get("index").getDesc() + ", "
                + super.getUInt("dimensions"));
    }
    
}

File: src/main/java/com/github/zxh/classpy/classfile/bytecode/Multianewarray.kt
package com.github.zxh.classpy.classfile.bytecode

import com.github.zxh.classpy.classfile.reader.ClassReader

class Multianewarray(opcode: Opcode, pc: Int) : Instruction(opcode, pc) {

    init {
        u1  ("opcode");
        u2cp("index");
        u1  ("dimensions");
    }

    override fun readOperands(reader: ClassReader) {
        val index = super.get("index").desc
        val dimensions = super.getInt("dimensions")
        desc = "${opcode.name} $index, $dimensions"
    }
    
}
